.changeset/light-games-clap.md

```diff
@@ -0,0 +1,6 @@
+---
+'@redocly/openapi-core': minor
+'@redocly/cli': minor
+---
+
+Fixed an issue where remove-unused-components would not remove schemas which had a top-level allOf and improved detection of unused schemas in no-unused-components rule.
```

Comment:
Could you rephrase the description so it emphasises the improvement rather than the fix?